triplea-game/triplea
# Move remaining lobby server code to lobby project

## Overview

Moves the remaining lobby server code from the `game-core` project to the `lobby` project.  This should be the last PR related to moving lobby-specific code to the `lobby` project.

## Functional Changes

None.

## Refactoring Changes

* When I moved `LobbyPropertyReader`, I changed its package from `g.s.engine.config.lobby` to `g.s.engine.lobby.server.config` to be more consistent with the names of the other packages in the `lobby` project.
* The `g.s.test.Assertions` class was moved to `o.t.test.common` in the `test-common` project so it can be used by both the `game-core` and `lobby` projects.

## Manual Testing Performed

Smoke tested the `lobby` project build artifact to ensure I could:

* Connect a lobby client
* Connect a headless game server
* Join a headless game server
* Mute lobby users
